src/sw/redis++/redlock.hpp

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************
+   Copyright (c) 2019
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ *************************************************************************/
+
+#ifndef REDISPLUSPLUS_REDLOCK_H
+#define REDISPLUSPLUS_REDLOCK_H
+
+#include "redis++.h"
+#include <unistd.h>
+#include <openssl/rc4.h>
+
+namespace sw {
+
+namespace redis {
+
+class RandomBufferInterface
```

Comment:
You can use @antirez's method to generate random strings automatically.

Reply:
I am still sticking with the `RandomBufferInterface`, as I think the user should be able to specify the `random` algorithm. It should be up to the user to decide if they want more speed or more randomness. The implementation is only in the test cases, and there I used the now static `RedLock::lock_id()`, as algorithm for the non-openssl case. See PR #33.